packages/react-instantsearch/src/components/SearchBox.js

```diff
@@ -120,6 +120,7 @@ class SearchBox extends Component {
   onSubmit = e => {
     e.preventDefault();
     e.stopPropagation();
+    this.input.blur();
```

Comment:
although this *is* a breaking change in a way, a user expects something to happen when pressing "search". You would need to tab back into the input field, but that's how it is in other search that doesn't do search as you type (amazon for ex)